### PR TITLE
DLS-3403 Added audit events

### PR DESCRIPTION
**Pull request by gavinharris43**

### CODE REVIEW

**Review by lutnos — COMMENTED** [review]
... plus I think we may have a problem in Travis, the job has bee queued for about an hour.  Not sure what to do about that.  Let's hope the problem goes away, say overnight

**Review by lutnos — CHANGES_REQUESTED** [review]
Gavin, my earlier Comment was meant to be a Request changes!

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala [hunk 1]
```diff
@@ -163,6 +164,10 @@ class ReminderController @Inject() (
                             true,
                             daysToReceiveReminders
                           )
+                          auditor.sendEvent(
+                            HtsReminderUserCreatedEvent(HtsReminderUserCreated(Json.toJson(htsUserToBeUpdated)), request.uri),
```
[the remark this answers is not in the document]
Comment (gavinharris43) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: done

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala [hunk 2]
```diff
@@ -287,6 +292,11 @@ class ReminderController @Inject() (
                       emailRetrieved match {
                         case Some(email) if !email.isEmpty ⇒ {
                           if (success.reminderFrequency === "cancel") {
+                            auditor.sendEvent(
+                              HtsReminderCancelled(HtsReminderUserCancelled(
+                                userInfo.nino, userInfo.email.getOrElse("")), request.uri),
```
[the remark this answers is not in the document]
Comment (gavinharris43) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: done

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala [hunk 3]
```diff
@@ -311,6 +321,10 @@ class ReminderController @Inject() (
                               true,
                               daysToReceiveReminders
                             )
+                            auditor.sendEvent(
+                              HtsReminderUserUpdatedEvent(HtsReminderUserUpdated(Json.toJson(htsUserToBeUpdated)), request.uri),
```
[the remark this answers is not in the document]
Comment (gavinharris43) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: done

**Review by lutnos — APPROVED** [review]

**Pull request merged**